splice_machine/src/main/java/com/splicemachine/derby/stream/function/csv/CsvLineReaderCR.java

```diff
@@ -0,0 +1,194 @@
+/*
+ * Copyright (c) 2012 - 2020 Splice Machine, Inc.
+ *
+ * This file is part of Splice Machine.
+ * Splice Machine is free software: you can redistribute it and/or modify it under the terms of the
+ * GNU Affero General Public License as published by the Free Software Foundation, either
+ * version 3, or (at your option) any later version.
+ * Splice Machine is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Affero General Public License for more details.
+ * You should have received a copy of the GNU Affero General Public License along with Splice Machine.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.splicemachine.derby.stream.function.csv;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+import java.io.IOException;
+import java.io.Reader;
+
+/**
+ * A buffered CSV line reader which is able to return the current line ending's type via
+ * getCurrentLineEndingType / getCurrentLineEnding (like "\r\n", "\r", "\n", or "" = EOF)
+ */
+public class CsvLineReaderCR {
+    private final Reader reader;
+    private final int configInitialStringBufferSize;
+    private final int configBufferSize;
+
+    private char bufferBytes[];
+    private int bufferSize, bufferPos;
+    private int lineNumber = 0;
+
+    private StringBuffer res = null;
```

Comment:
We should use StringBuilder (non thread safe version, recommended)

Reply:
ah ok

Reply:
done